beginner-challenges/challenge-2/index.js

```diff
@@ -0,0 +1,31 @@
+
+function handleUserInput() {
+    let num = document.getElementById('inputNumber').value;
+    num = parseInt(num);
+
+    if (num < 1 || isNaN(num)) {
+        alert("Please enter a valid non-negative number.");
+        return;
+    }
+
+    let tableDiv = document.getElementById('tableDiv');
+
+    // Clear previous results
+    tableDiv.innerHTML = '';
+
+    for (let i = 1; i <=100; i++) {
+        const rowDiv = document.createElement('div');
```

Comment:
The general idea is to use a table, tr and td elements.

When your number is 5 the table will have 5 columns and 5 rows.

On position (0,0) the value is 1. On the position (5,5) the value is 25. on (3,4) the value is 12.